### PR TITLE
Improved connection handshake session

### LINKED ISSUE #1422: Improve connection speed by running fewer commands
Today, connecting to a Redis instance, involves running multiple commands, and parsing the response repeatedly, this includes usage such as [setinfo](https://redis.io/commands/client-setinfo), which itself is run twice. The example below illustrates, that it's possible to pass all connection configuration information within a pipeline. In turn, the HELLO call returns the version of Redis, which could perhaps be used later for other things.

This is marked as _good-first-issue_ not because it's necessarily easy to implement, inasmuch as it's a good entrypoint, to learning how connections to redis occur.

```
127.0.0.1:6379> multi
OK
127.0.0.1:6379(TX)> auth bob smurf
QUEUED
127.0.0.1:6379(TX)> select 5
QUEUED
127.0.0.1:6379[5](TX)> hello 3
QUEUED
127.0.0.1:6379[5](TX)> client setinfo lib-name predis
QUEUED
127.0.0.1:6379[5](TX)>  client setinfo lib-ver 55.5.0
QUEUED
127.0.0.1:6379[5](TX)> client tracking on optin
QUEUED
127.0.0.1:6379[5](TX)> CLIENT caching yes
QUEUED
127.0.0.1:6379[5](TX)> exec
1) OK
2) OK
3) 1# "server" => "redis"
   2# "version" => "7.2.3"
   3# "proto" => (integer) 3
   4# "id" => (integer) 26
   5# "mode" => "standalone"
   6# "role" => "master"
   7# "modules" => (empty array)
4) OK
5) OK
6) OK
7) OK
127.0.0.1:6379[5]> 
```

### PR DESCRIPTION
**Pull request by vladvildanov**
Closes #1422 

This PR changes on-connection behaviour by using fewer commands during handshake

Additional enhancements:
- Updated `CommandInterface` with `serializeCommand()` method that returns RESP representation of command. According to Information Expert pattern Command object should provides it's serialized representation instead of Connection object.
- Updated `NodeConnectionInterface` with `write()` method that writes serialized command directly into resource. This method existed before but was protected.
- Updated `NodeConnectionInterface` with `getClientId()` method that returns CLIENT ID assigned by Redis server for current connection.

### CODE REVIEW

**Comment by vladvildanov** [comment]
@tillkruss This PR is blocked by #1437 so it doesn't look like a total mess

**Comment by tillkruss** [comment]
Can we rebase after merging the styling changes first? Hard to read otherwise.

**Comment by vladvildanov** [comment]
@tillkruss Done!

**Review by tillkruss — APPROVED** [review]

**Review by tillkruss — APPROVED** [review]

**Review by tillkruss — APPROVED** [review]

**Pull request merged**
